src/NUnitEngine/nunit.engine.core/Services/ExtensionManager.cs

```diff
@@ -233,14 +255,35 @@ private ExtensionPoint DeduceExtensionPointFromType(TypeReference typeRef)
         }
 
         /// <summary>
-        /// Find candidate extension assemblies starting from a
-        /// given base directory.
+        /// Find candidate extension assemblies starting from a given base directory
+        /// and using the contained '.addins' files to direct the search.
         /// </summary>
-        /// <param name="startDir"></param>
-        private void FindExtensionAssemblies(IDirectory startDir)
+        /// <param name="initialDirectory">Path to the initial directory.</param>
+        private void FindExtensionAssemblies(IDirectory initialDirectory)
         {
-            // First check the directory itself
-            ProcessAddinsFiles(startDir, false);
+            // Since no patterns are provided, look for addins files
+            ProcessAddinsFiles(initialDirectory, false);
+        }
+
+        /// <summary>
```

Comment:
Should we not use `<inheritdoc />` to prevent risking interface and implementation documentation getting out of sync?

Reply:
In my own code I have always avoided it but I know others use it. For me it's a question of compatibility. If somebody wants to use a different compiler or ide, do you know if will `<inhreitdoc>` always work? In any case, if we decided to always use it, then I'd do one big PR rather than changing things one at a time. What do you think?

Reply:
The compiler does the copying to the XML file.
See [inheritdoc](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/xmldoc/recommended-tags#inheritdoc)

Reply:
The _Microsoft_ compilers do. But maybe I'm beating a dead horse. Currently, I can't find any viable alternatives to the MS compiler, such as we used to have a long time ago. And of course we're using it to build the engine anyway.

I don't like to combine unrelated changes in the same PR, so I'll create an issue.